### PR TITLE
[UWP] Add role property to actions

### PR DESCRIPTION
**Pull request by anna-dingler**
# Related Issue

Progress towards #8041

# Description

Added a new property `role` to actions. The new property will allow the card author to specify an aria role for a buttons and improve their card's accessibility.

Proposal can be found here: https://github.com/microsoft/AdaptiveCards/issues/8041#issuecomment-1438924387

# Sample Card

Added a sample card for v1.6

# How Verified

Verified manually in the Visualizer with Narrator.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/8337)

### CODE REVIEW

**Comment by anna-dingler** [comment]
@jwoo-msft @almedina-ms Do we need any changes for Android or iOS here?

**Review by jwoo-msft — APPROVED** [review]

File: source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/EverythingBagel.h [hunk 1]
```diff
@@ -2,6 +2,6 @@
...
...sociatedInputs\":\"None\",\"id\":\"ActionSet.Action.Submit_id\",\"isEnabled\":false,\"title\":\"ActionSet.Action.Submit\",\"tooltip\":\"tooltip\",\"type\":\"Action.Submit\"},{\"id\":\"ActionSet.Action.OpenUrl_id\",\"role\":\"Link\",\"title\":\"ActionSet.Action.OpenUrl\",\"tooltip\":\"tooltip\",\"type\":\"Action.OpenUrl\",\"url\":\"https://adaptivecards.io/\"}],\"type\":\"ActionSet\"},{\"horizontalAlignment\":\"right\",\"id\":\"RichTextBlock_id\",\"inlines\":[{\"color\":\"Dark\",\"fontType\":\"Monospace\",\"highlight\":true,\"isSubtle\":true,\"italic\":true,\"size\":\"Large\",\"strikethrough\":true,\"text\":\"This is a text run\",\"type\":\"TextRun\",\"underline\":true,\"weight\":\"Bolder\"},{\"selectAction\":{\"type\":\"Action.Submit\"},\"text\":\"This is another text run\",\"type\":\"TextRun\"},{\"text\":\"This is a text run specified as a string\",\"type\":\"TextRun\"}],\"type\":\"RichTextBlock\"}],\"fallbackText\":\"fallbackText\",\"lang\":\"en\",\"refresh\":{\"action\":{\"id\":\"refresh_action_id\",\"type\":\"Action.Execute\",\"verb\":\"refresh_action_verb\"},\"userIds\":[\"refresh_userIds_0\"]},\"rtl\":false,\"speak\":\"speak\",\"type\":\"AdaptiveCard\",\"version\":\"1.0\"}\n"
```
Comment (beervoley) [inline source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/EverythingBagel.h:5]: hard to see the diff here - what is it?:)

File: source/shared/cpp/ObjectModel/BaseActionElement.cpp [hunk 2]
```diff
@@ -134,6 +145,11 @@ Json::Value BaseActionElement::SerializeToJsonValue() const
         root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::IsEnabled)] = m_isEnabled;
     }
 
+    if (m_role != ActionRole::Button)
```
Comment (beervoley) [inline source/shared/cpp/ObjectModel/BaseActionElement.cpp:148]: why this check here?:)

Comment (anna-dingler) [reply source/shared/cpp/ObjectModel/BaseActionElement.cpp:148]: Button is the default so there's no need to update it

Comment (beervoley) [reply source/shared/cpp/ObjectModel/BaseActionElement.cpp:148]: gotcha

**Pull request merged**